dpi/src/lib.rs

```diff
@@ -143,21 +143,18 @@ impl<P> LogicalUnit<P> {
 }
 
 impl<P: Pixel> LogicalUnit<P> {
-    #[inline]
     pub fn from_physical<T: Into<PhysicalUnit<X>>, X: Pixel>(
         physical: T,
         scale_factor: f64,
     ) -> Self {
         physical.into().to_logical(scale_factor)
     }
 
-    #[inline]
     pub fn to_physical<X: Pixel>(&self, scale_factor: f64) -> PhysicalUnit<X> {
         assert!(validate_scale_factor(scale_factor));
         PhysicalUnit::new(self.0.into() * scale_factor).cast()
     }
 
-    #[inline]
     pub fn cast<X: Pixel>(&self) -> LogicalUnit<X> {
         LogicalUnit(self.0.cast())
```

Comment:
Small public function calling public code? All of these, it seems like?

Reply:
Emphasis on **non-generic**.

Reply:
Let's wake up and try reading this again after a few hours :see_no_evil:.